pkg/providers/imagefamily/agentbakerbootstrap/provisionclientbootstrap.go

```diff
@@ -0,0 +1,239 @@
+	}
+
+	switch p.ImageFamily {
+	case v1alpha2.Ubuntu2204ImageFamily:
+		provisionProfile.OsSku = to.Ptr(models.OSSKU_Ubuntu)
+	case v1alpha2.AzureLinuxImageFamily:
+		provisionProfile.OsSku = to.Ptr(models.OSSKU_AzureLinux)
+	default:
+		provisionProfile.OsSku = to.Ptr(models.OSSKU_Ubuntu)
+	}
+
+	if p.KubeletConfig != nil {
+		provisionProfile.CustomKubeletConfig = &models.CustomKubeletConfig{
+			// AllowedUnsafeSysctls: ..., // Unsupported as of now
+			CPUCfsQuota: p.KubeletConfig.CPUCFSQuota,
+		}
+	}
+
+	if p.KubeletConfig != nil && p.KubeletConfig.MaxPods != nil {
+		provisionProfile.MaxPods = p.KubeletConfig.MaxPods
+	} else {
+		provisionProfile.MaxPods = lo.ToPtr(int32(250)) // Delegatable defaulting?
+	}
+
+	if modeString, ok := p.Labels["kubernetes.azure.com/mode"]; ok && modeString == "system" {
+		provisionProfile.Mode = lo.ToPtr(models.AgentPoolMode_System)
+	} else {
+		provisionProfile.Mode = lo.ToPtr(models.AgentPoolMode_User)
+	}
+
+	if utils.IsNvidiaEnabledSKU(p.InstanceType.Name) {
+		if utils.UseGridDrivers(p.InstanceType.Name) {
+			provisionProfile.GpuProfile = &models.GPUProfile{
+				DriverType:       lo.ToPtr(models.DriverType_GRID),
+				InstallGPUDriver: lo.ToPtr(true),
+			}
+		} else {
+			provisionProfile.GpuProfile = &models.GPUProfile{
+				DriverType:       lo.ToPtr(models.DriverType_CUDA),
+				InstallGPUDriver: lo.ToPtr(true),
+			}
+		}
+
+	}
+
+	if p.IsWindows {
+		provisionProfile.OsType = lo.ToPtr(models.OSType_Windows)
+	} else {
+		provisionProfile.OsType = lo.ToPtr(models.OSType_Linux)
+	}
+
+	provisionHelperValues := &models.ProvisionHelperValues{
+		SkuCPU:    lo.ToPtr(p.InstanceType.Capacity.Cpu().AsApproximateFloat64()),
+		SkuMemory: lo.ToPtr(p.InstanceType.Capacity.Memory().AsApproximateFloat64()),
+	}
+
+	return p.getNodeBootstrappingFromClient(ctx, provisionProfile, provisionHelperValues, p.KubeletClientTLSBootstrapToken)
+}
+
+func (p *ProvisionClientBootstrap) getNodeBootstrappingFromClient(ctx context.Context, provisionProfile *models.ProvisionProfile, provisionHelperValues *models.ProvisionHelperValues, bootstrapToken string) (string, string, error) {
+	transport := httptransport.New(options.FromContext(ctx).NodeBootstrappingServerURL, "/", []string{"http"})
+	client := client.New(transport, strfmt.Default)
+
+	params := operations.NewNodeBootstrappingGetParams()
+	params.ResourceGroupName = p.ClusterResourceGroup
+	params.ResourceName = p.ClusterName
+	params.SubscriptionID = p.SubscriptionID
+	provisionValues := &models.ProvisionValues{
+		ProvisionProfile:      provisionProfile,
+		ProvisionHelperValues: provisionHelperValues,
+	}
+	params.Parameters = provisionValues
+
+	params.WithTimeout(30 * time.Second)
+	params.Context = ctx
+
+	resp, err := client.Operations.NodeBootstrappingGet(params)
+	if err != nil {
+		return "", "", err
+	}
+
+	if resp.Payload == nil {
+		return "", "", fmt.Errorf("no payload in response")
+	}
+	if resp.Payload.Cse == nil || *resp.Payload.Cse == "" {
```

Comment:
nit: CSE should be cased consistently across the code

Reply:
Agreed, though some of this may be coming from Swagger etc.

Reply:
All this come un-golang namings from the auto-generated client.